src/util/mpir_hwtopo.c

```diff
@@ -201,23 +201,39 @@ int MPII_hwtopo_init(void)
     bindset_is_valid = 0;
 
 #ifdef HAVE_HWLOC
+    int status = -1;
     bindset = hwloc_bitmap_alloc();
-    hwloc_topology_init(&hwloc_topology);
-    char *xmlfile = MPIR_pmi_get_jobattr("PMI_hwloc_xmlfile");
-    if (xmlfile != NULL) {
-        int rc;
-        rc = hwloc_topology_set_xml(hwloc_topology, xmlfile);
-        if (rc == 0) {
-            /* To have hwloc still actually call OS-specific hooks, the
-             * HWLOC_TOPOLOGY_FLAG_IS_THISSYSTEM has to be set to assert that the loaded
-             * file is really the underlying system. */
-            hwloc_topology_set_flags(hwloc_topology, HWLOC_TOPOLOGY_FLAG_IS_THISSYSTEM);
-        }
-        MPL_free(xmlfile);
+
+#if ENABLE_PMIX
```

Comment:
How about wrapping it inside MPIR_pmi abstraction as `MPIR_pmi_load_hwloc_topology`?